lib/ASTDecl.cpp

```diff
@@ -98,29 +98,71 @@ namespace pillar
       std::string name = np.VariableName(var_decl_op);
       mlir::Type varType = var_decl_op.getType();
       // // Create Clang QualType from MLIR Type
-      clang::QualType clangType = LiftType(varType);
-      clang::VarDecl *var_decl = CreateVarDeclFromStrRef(sdc, ldc, clangType, name);
+      clang::QualType clang_type = LiftType(varType);
+      clang::VarDecl *var_decl = CreateVarDeclFromStrRef(sdc, ldc, clang_type, name);
       sdc->addDecl(var_decl);
       op_to_decl.emplace(
           var_decl_op,
           var_decl);
-      mlir::Region *init = &(var_decl_op.getInitializer());
+      if (mlir::Region *init = &(var_decl_op.getInitializer()))
+      {
 
-      AddToLiftQueue([=, this]()
-                     {
+        AddToLiftQueue([=, this]()
+                       {
       for (mlir::Block &block : init->getBlocks())
       {
         var_decl->setInit(LiftBlockExpr(sdc, block));
         break;
       } });
+      }
 
       return var_decl;
     }
-    void AST::LiftTypeDefOp(clang::DeclContext *sdc,
-                            clang::DeclContext *ldc,
-                            vast::hl::TypeDefOp type_def_op)
+    clang::FieldDecl *AST::LiftFieldDeclOp(clang::DeclContext *sdc,
+                                           clang::DeclContext *ldc, clang::RecordDecl *record,
+                                           vast::hl::FieldDeclOp field_decl_op)
+    {
+
+      // std::string name = np.FieldName(field_decl_op);
+      std::string name = field_decl_op.getName().str();
```

Comment:
Add a method to that API to get names for a given `vast::hl::FieldDeclOp`, and default-implement the API as you have above.